src/main/java/socket/BidWS.java

```diff
@@ -353,6 +379,155 @@ protected void onAuctionBid(BodyWS body) {
         auctionBidded(dbBid);
     }
 
+    private void onCombinatorialAuctionBid(BodyWS body, Bid[] unsafeBids, Auction dbAuction, int userId) {
+        BidDAO bidDAO = BidDAOSQL.getInstance();
+        UserDAO userDAO = UserDAOSQL.getInstance();
+        GoodDAO goodDAO = GoodDAOSQL.getInstance();
+
+        List<Bid> unsafeBidsList = new ArrayList<>(Arrays.asList(unsafeBids));
+
+        // 1. All bids must have the same auction ID and amount.
+        double commonAmount = Math.floor(unsafeBids[0].amount * 100) / 100;
+        int commonAuctionId = unsafeBids[0].auctionId;
+        Set<Integer> differentGoods = new HashSet<>();
+        for (Bid unsafeBid: unsafeBidsList) {
+            unsafeBid.amount = Math.floor(unsafeBid.amount * 100) / 100;
+            if (unsafeBid.amount <= 0.1) {
```

Comment:
Should be 1.0 (Min) in combinatorial, yes?

Reply:
Yes.